SciSpaCy/custom_sentence_segmenter.py

```diff
@@ -1,19 +1,21 @@
 
+from typing import List, Optional, Any
 from SciSpaCy.consts import ABBREVIATIONS # pylint: disable-msg=E0611,E0401
+from spacy.tokens import Doc, Token
 
-def combined_rule_sentence_segmenter(doc):
+def combined_rule_sentence_segmenter(doc: Doc) -> Doc:
     """Adds sentence boundaries to a Doc. Intended to be used as a pipe in a spaCy pipeline.
 
     @param doc: the spaCy document to be annotated with sentence boundaries
     """
 
     # keep track of the two previous tokens
-    prev_tokens = [None, None]
+    prev_tokens: List[Any] = [None, None]
```

Comment:
Try not to use `Any`. I think these should be `spacy.Token`? It's fine to declare their type in the type hint but then leave it as `None` in the assignment, you're basically just saying "anything I put in here _will_ have this type"

Reply:
So they should be tokens but then the type checker fails because they are assigned to None, and i try to access the .text attribute of them and it says None has no attribute text. Is there another way around this?

Reply:
I tried Optional[Any] because it seemed like that was the equivalent of a nullable type, but it didn't like that either

Reply:
I think if you separate them it will work, e.g `token1: Token = None, token2: Token = None`

Reply:
Thanks, that seems to have worked